broker/graphite/src/macro_cache.cc

```diff
@@ -142,67 +131,42 @@ std::string const& macro_cache::get_instance(uint64_t instance_id) const {
   if (found == _instances.end())
     throw msg_fmt("graphite: could not find information on instance {}",
                   instance_id);
-  return found->second->type() == neb::instance::static_type()
-             ? std::static_pointer_cast<neb::instance>(found->second)->name
-             : std::static_pointer_cast<neb::pb_instance>(found->second)
-                   ->obj()
-                   .name();
+  return std::static_pointer_cast<neb::pb_instance>(found->second)
+      ->obj()
+      .name();
```

Comment:
the _instances table can be changed to only accept pb_instances. So the cast should be no more needed.